src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/SunPKCS11.java

```diff
@@ -1073,6 +1079,17 @@ private static void register(Descriptor d) {
                 m(CKM_TLS_PRF, CKM_NSS_TLS_PRF_GENERAL));
         d(KG, "SunTls12Prf", "sun.security.pkcs11.P11TlsPrfGenerator",
                 m(CKM_TLS_MAC));
+
+        d(KDF, "HKDF-SHA1", P11KDF, m(CKM_SHA_1_HMAC),
+                m(CKM_HKDF_DERIVE, CKM_HKDF_DATA));
+        d(KDF, "HKDF-SHA224", P11KDF, m(CKM_SHA224_HMAC),
+                m(CKM_HKDF_DERIVE, CKM_HKDF_DATA));
+        d(KDF, "HKDF-SHA256", P11KDF, m(CKM_SHA256_HMAC),
+                m(CKM_HKDF_DERIVE, CKM_HKDF_DATA));
+        d(KDF, "HKDF-SHA384", P11KDF, m(CKM_SHA384_HMAC),
+                m(CKM_HKDF_DERIVE, CKM_HKDF_DATA));
+        d(KDF, "HKDF-SHA512", P11KDF, m(CKM_SHA512_HMAC),
+                m(CKM_HKDF_DERIVE, CKM_HKDF_DATA));
```

Comment:
We only defined HKDF-SHA256 and later in the Java Security Standard Names doc.

Reply:
We included SHA1 because there could be a legacy use case to support and it's part of the test vectors for RFC 5869 (HMAC-based Extract-and-Expand Key Derivation Function (HKDF)). We don't recommend using it, and will probably filter it out once we have the Filter integrated, but would you be okay with keeping it?

Reply:
Do you have any data on how many legacy uses case use it? I think for new mechanisms we should be forward looking and refrain from adding support for weak or not recommended algorithms unless there is a very good reason. It is often harder to remove something than to add it.

Reply:
Yes, I see your point. We don't have data to back it up. If someone comes up with a strong case, we can reconsider it in the future. We removed it.